### ISSUE TITLE
Allow integration with `dbplyr`

### ISSUE THREAD

**Issue by markfairbanks** [markfairbanks/tidytable#748]
Basically this should be possible:
``` r
library(dbplyr)
library(tidytable)

memdb_frame(x = 1:3, y = c("a", "a", "b")) %>%
  mutate(x = x * 2) %>%
  collect() %>%
  summarize(avg_x = mean(x), .by = y)
#> # A tidytable: 2 × 2
#>   y     avg_x
#>   <chr> <dbl>
#> 1 a         3
#> 2 b         6
```

**Comment by dibbles21** [comment]
HI Mark. I'm curious whether this issue would potentially allow the issue [here](https://github.com/markfairbanks/tidytable/issues/730) to also be solved?

**Comment by markfairbanks** [comment]
Yes it would - I'm not 100% sure I want to do this, but it's something I'm considering.

The positive is it would allow integration with `dbplyr`, the negative is I would need to start considering whether to build in integrations with other packages.

Here's a small list of packages that I would have to consider:
* `dbplyr`
* `arrow`
* `tsibble`
* `fable`

...and I'm guessing there are more I'm not thinking of.

**Comment by dibbles21** [comment]
Thank you for confirming. I'll keep an eye on this issue as if resolved it's something I'd be keen on using.

**Comment by markfairbanks** [comment]
I think the most likely scenario is I create a `tidytable` extension package (I have no idea what the name would be) that would add in these integrations
